src/main/resources/config/application.yml

```diff
@@ -208,3 +208,7 @@ application:
             max-attempts: 10
             delay: 100 #in milliseconds
             multiplier: 2
+    load-balanced-rest-template:
+        connection-request-timeout: 0
```

Comment:
set by default correct values (that will not bloke existing systems)

Reply:
Assert.isTrue(timeout >= 0, "Timeout must be a non-negative value");
it is correct, only negative values block existing systems